library/core/src/sync/atomic.rs

```diff
@@ -3326,23 +3326,23 @@ macro_rules! atomic_int {
                 unsafe { atomic_xor(self.v.get(), val, order) }
             }
 
-            /// Fetches the value, and applies a function to it that returns an optional
-            /// new value. Returns a `Result` of `Ok(previous_value)` if the function returned `Some(_)`, else
-            /// `Err(previous_value)`.
+            /// Loads the current value, applies a closure to it, and optionally tries to store a new value.
             ///
-            /// Note: This may call the function multiple times if the value has been changed from other threads in
-            /// the meantime, as long as the function returns `Some(_)`, but the function will have been applied
-            /// only once to the stored value.
+            /// If the closure ever returns None, this method will immediately return `Err(current value)`.
+            /// If the closure returns Some(new value), then this method calls
+            #[doc = concat!("[`", stringify!($atomic_type), "::compare_exchange_weak`]")]
+            /// to try to store the new value.
```

Comment:
Perhaps explain the semantics first, I do think a callout to the implementation is appropriate in parens here though.

> If the closure returns `Some(updated_value)`, the implementation will attempt to store the `updated_value` to the atomic. This store will only occur if the value of the atomic is still equal to the value of the initial load (see `compare_exchange_weak`).